### PR TITLE
PopulationManager class #23

### PR DESCRIPTION
**Pull request by SlowFaller**
stubbed out with basic functionality #23

### CODE REVIEW

File: scripts/systems/population_manager.gd [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+const MIN_POPULATION = 3
+const MAX_POPULATION = 50
+const MIN_HUNGER = 0
+const MAX_HUNGER = 100
+const MIN_FULFILLMENT = 0
+const MAX_FULFILLMENT = 100
+
+@export_group("Starting values")
+@export var starting_population: int = MIN_POPULATION
+@export var starting_housed: int = MIN_POPULATION
```
Comment (JakeGilla-DL) [inline scripts/systems/population_manager.gd:12]: This should be a separate constant.

File: scripts/systems/population_manager.gd [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+@export var starting_hunger: int = MIN_HUNGER
```
Comment (JakeGilla-DL) [inline scripts/systems/population_manager.gd:13]: Starting Hunger should be a seperate constant

File: scripts/systems/population_manager.gd [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+@export var starting_fulfillment: int = MIN_FULFILLMENT
```
Comment (JakeGilla-DL) [inline scripts/systems/population_manager.gd:14]: starting fulfillment should be a seperate constants

File: scripts/systems/population_manager.gd [hunk 4]
```diff
@@ -0,0 +1,76 @@
+extends Node
+
+const MIN_POPULATION = 3
+const MAX_POPULATION = 50
+const MIN_HUNGER = 0
+const MAX_HUNGER = 100
+const MIN_FULFILLMENT = 0
+const MAX_FULFILLMENT = 100
+
+@export_group("Starting values")
```
Comment (JakeGilla-DL) [inline scripts/systems/population_manager.gd:3]: starting values should be their own constants: for example we might want min/max hunger to be 0/100 but to start with 30 hunger.

File: scripts/systems/population_manager.gd [hunk 5]
```diff
@@ -0,0 +1,76 @@
+extends Node
+
+const MIN_POPULATION = 3
```
Comment (JakeGilla-DL) [inline scripts/systems/population_manager.gd]: We should move all these constants and the starting values below to a new resource, the read those resources in as a part of initialization

File: scripts/systems/population_manager.gd (hunk 4 above)
Comment (JakeGilla-DL) [reply scripts/systems/population_manager.gd:3]: Once these are in a resource, these wont need to be export vars any more

Comment (JakeGilla-DL) [reply scripts/systems/population_manager.gd:3]: Also the init should probably assert that any starting value isnt less than the minimum or greater than the maximum.

**Review by JakeGilla-DL — CHANGES_REQUESTED** [review]

File: scripts/systems/population_manager.gd (hunk 4 above)
Comment (SlowFaller) [reply scripts/systems/population_manager.gd:3]: Done

File: scripts/systems/population_manager.gd (hunk 5 above)
Comment (SlowFaller) [reply scripts/systems/population_manager.gd]: Done

**Review by JakeGilla-DL — APPROVED** [review]

**Review by SlowFaller — COMMENTED** [review]
Looks good,

**Pull request merged**